src/Workspaces/Core/Portable/Options/IGlobalOptionService.cs

```diff
@@ -18,6 +19,11 @@ internal interface IGlobalOptionService : IOptionsReader
     /// </summary>
     T GetOption<T>(Option2<T> option);
 
+    /// <summary>
+    /// Gets the current value of the specific option.
+    /// </summary>
+    ValueTask<T> GetOptionAsync<T>(Option2<T> option);
```

Comment:
take a CT?

Reply:
Yeah, probably better to enforce sending over a CT from the get-go.